e2e/test/multiaz/provider.go

```diff
@@ -55,7 +55,7 @@ func NewProvider(config ProviderConfig) (*Provider, error) {
 }
 
 func (p *Provider) GetClusterAZs(ctx context.Context) ([]string, error) {
-	vmss, err := p.azureClient.VirtualMachineScaleSetsClient.Get(ctx, p.clusterID, fmt.Sprintf("%s-worker-%s", p.clusterID, p.clusterID))
+	vmss, err := p.azureClient.VirtualMachineScaleSetsClient.Get(ctx, p.clusterID, fmt.Sprintf("nodepool-%s", p.clusterID))
```

Comment:
Would it be useful to have this VMSS name saved as a annotation on `AzureMachinePool`?

Reply:
Storing everything in the CRs, being explicit and all that :smile:

Reply:
Then validating the VMSS name in a webhook, i.e. implementing there a convention that it is called `nodepool-<cluster id>`.

Reply:
Are you saying that all `key` functions could be converted to annotations?

![](https://24.media.tumblr.com/tumblr_me2mheoM8z1rl2z7po1_250.gif)

Reply:
Hahah, well, I guess that most of them have a decent potential to be converted into an annotation (or a label, or something else), which can be then validated and set to a default value if necessary.

Basically:
- Storing data in the CRs.
- Defining domain/business rules around data in the webhooks, so that nothing invalid gets into the system. When we change the rules, we change them here.